src/oph/ehoks/db/postgresql.clj

```diff
@@ -575,3 +579,110 @@
   (query
     [queries/select-hankitun-osaamisen-naytot-by-yto-osa-alue-id id]
     {:row-fn h/hankitun-osaamisen-naytto-from-sql}))
+
+(defn delete-hoks-or-part! [t id_name id_value]
+  (let [query (str id_name " = " id_value)]
+    (jdbc/delete! {:connection-uri (:database-url config)} t (vector query))))
+
+(defn delete-group! [delete-map ids]
+  (doseq [[k v] delete-map]
+    (let [delete-response (map #(delete-hoks-or-part! k v %) ids)
+          parsed-response
+          (mapv #(first (nth delete-response %)) (range (count ids)))]
+      (log/info "Deleted :" (reduce + parsed-response) " resources"))))
+
```

Comment:
Tästä kommentista alaspäin oikeastaan kaikissa funktioissa on ongelmia omaan silmään:
+ Funktioissa ei hyödynnetä tietokantaa ollenkaan. Poisto ja datan käsittely tehdään koodissa, mistä tulee ylimääräisiä kutsuja sekä datan hakemisessa ja datan poistossa.
  + Tässä voisi hyödyntää yksinkertaisesti sitä tietoa, että jokaisella hoksin osalla on hoksin id, millä arvon voi poistaa. Riittää, että käyttää `CASCADE`a, niin saa poistettua HOKSin ja sen kaikki arvot syvemmältä ilman erillisiä kutsuja.
+ Miksi muuttujissa ja parametreissa käytetään alaviivoja väliviivan sijaan?